## *Syllabus*

# Comparative Genomics

	Yale University, Spring 2019
	E&EB 723

	Time: Thursday 1-3
	Location: ESC 100
	
	Instructor: Casey Dunn
	Prepend the subject line of all course related emails with "genomics: "


The field of evolutionary biology is increasingly drawing on genomic data. The field of genomic biology is becoming more evolutionary as genomes are sequenced for a broader diversity of organisms. This course focuses on the evolution of genome sequence and function at macroevolutionary timescales, with an emphasis on building practical computational skills for genomic and phylogenetic comparative analyses. There will be more focus on using phylogenies to understand genome evolution than on using genomes to build phylogenies.

## Technical details

### Git

This course is organized with [github education tools](https://education.github.com/guide). The course will make heavy use of [git](https://git-scm.com/) for sharing, communication, and collaboration. All students need to have a github account, preferably one that is registered to their Yale email address so that they get the [full academic features](https://education.github.com/discount).

### HPC

The class will be given student cluster accounts on one of the Yale High Performance Compute (HPC) clusters. Access to both compute and storage resources will last the course of the semester, after which data need to be copied elsewhere if they are important enough to be saved.

## Course format

Classes will consist of lectures, student led discussions, and computational labs.

## Course site

All materials for the course, including the syllabus, are available at the [course site](https://github.com/Yale-EEB723). The syllabus will be updated as the course progresses, please check it weekly. Please submit suggestions and corrections for the class via the [issue tracker](https://github.com/Yale-EEB723/syllabus/issues).

## Projects

The class will work collaboratively on projects to assemble and annotate genomes, and then to analyze them in an evolutionary comparative context. Students will be leaders on subprojects within this larger collaborative framework.

We will focus on publicly available data, but if students have their own genome data we can also analyze those.


## Reading

Reading includes manuscripts, book chapters, online resources, and videos to be watched ahead of class. The dates the readings will be discussed in class are listed in the schedule, but some will be useful to you much earlier as you work on your projects. In addition, the reading load is very uneven. On light weeks, it is good to get a jump on reading for future weeks.

In addition to reading assigned for each class, the following will be used as references throughout the course:

Haddock, SHD and CW Dunn (2011). Practical Computing for Biologists. [amazon](http://www.amazon.com/Practical-Computing-Biologists-Steven-Haddock/dp/0878933913/ref=sr_1_1)

Whickham, H (2017). R for Data Science. http://r4ds.had.co.nz


## Schedule

### Week 1 - Intro to comparative genomics. Questions, methods, history

### Week 2 - Sequencing technology

### Week 3 - Practical computing skills

### Week 4 - Genome assembly

### Week 5 - Genome annotations

### Week 6 - Genome annotations

### Week 7 - Phylogenetic methods

### Week 8 - Phylogenetic comparative methods

### Week 9 - Genome evolution

### Week 10 - Functional genomics

### Week 11 - Comparative functional genomics

### Week 12 - Project

### Week 13 - Project
